actions/push/action.yaml

```diff
@@ -0,0 +1,168 @@
+  digest:
+    description: The digest of the OCI artifact. Empty if the artifact was not pushed.
+    value: ${{ steps.push.outputs.digest }}
+  digest-url:
+    description: The digest URL of the OCI artifact, i.e. including the repository URL. Empty if the artifact was not pushed.
+    value: ${{ steps.push.outputs.digest-url }}
+runs:
+  using: composite
+  steps:
+  - name: Check if the Flux CLI is installed
+    shell: bash
+    run: |
+      if ! flux version --client > /dev/null 2>&1; then
+        echo ""
+        echo "The Flux CLI is not installed. Please install it first using this GitHub Action:"
+        echo ""
+        echo "  https://github.com/controlplaneio-fluxcd/distribution/tree/main/actions/setup"
+        echo ""
+        exit 1
+      fi
+
+  - name: Check if the repository is valid
+    shell: bash
+    env:
+      REPOSITORY: ${{ inputs.repository }}
+    run: |
+      if [ "$REPOSITORY" != "$(echo $REPOSITORY | awk '{print tolower($1)}')" ]; then
+        echo ""
+        echo "The repository $REPOSITORY contains uppercase letters."
+        echo "OCI repositories must be lowercase, all the OCI tooling fails if they're not."
+        echo ""
+        exit 1
+      fi
+
+  - name: Define the --ignore-paths flag
+    id: ignore-paths
+    shell: bash
+    env:
+      IGNORE_PATHS: ${{ inputs.ignore-paths }}
+    run: |
+      ignore_paths_flag=""
+      if [ -n "$IGNORE_PATHS" ]; then
+        ignore_paths_flag="--ignore-paths=$IGNORE_PATHS"
+      fi
+
+      echo "flag=$ignore_paths_flag" >> $GITHUB_OUTPUT
+
+  - name: Diff the OCI artifact
+    id: diff
+    shell: bash
+    env:
+      REPOSITORY: ${{ inputs.repository }}
+      IGNORE_PATHS_FLAG: ${{ steps.ignore-paths.outputs.flag }}
+      INPUT_PATH: ${{ inputs.path }}
+      DIFF_TAG: ${{ inputs.diff-tag }}
+    run: |
+      pushed=true
+      if [ -n "$DIFF_TAG" ]; then
+        artifact_url=${REPOSITORY}:${DIFF_TAG}
+        if flux diff artifact oci://$artifact_url --path=$INPUT_PATH $IGNORE_PATHS_FLAG; then
+          pushed=false
+        fi
+      fi
+
+      echo "pushed=$pushed" >> $GITHUB_OUTPUT
+
+  - name: Push the OCI artifact
+    id: push
+    if: steps.diff.outputs.pushed == 'true'
+    shell: bash
+    env:
+      REPOSITORY: ${{ inputs.repository }}
+      IGNORE_PATHS_FLAG: ${{ steps.ignore-paths.outputs.flag }}
+      INPUT_PATH: ${{ inputs.path }}
+      DIFF_TAG: ${{ inputs.diff-tag }}
+      TAGS: ${{ inputs.tags }}
+    run: |
+      artifact_url=$REPOSITORY
+      if [ -n "$DIFF_TAG" ]; then
+        artifact_url=${REPOSITORY}:${DIFF_TAG}
+      elif [ -n "$TAGS" ]; then
+        IFS=',' read -ra tag_list <<< $TAGS
+        artifact_url=${REPOSITORY}:${tag_list[0]}
+      fi
+
+      echo "Pushing $INPUT_PATH to ${artifact_url}..."
+
+      push_output=$(flux push artifact oci://$artifact_url \
+        --source="$(git config --get remote.origin.url)" \
+        --revision="$(git branch --show-current)@sha1:$(git rev-parse HEAD)" \
```

Comment:
Would this work with tags? Don't we need to figure out if the checkout is for a tag and do `$(git tag --points-at HEAD)@sha1:$(git rev-parse HEAD)`

Reply:
I added this as the parameter `revision`.